hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -571,24 +613,22 @@ def onchange_unit_amount(self):
                 'title': _("Warning"),
                 'message': _("Save the Timesheet Sheet first.")
             }}
-        timesheets = sheet.timesheet_ids.filtered(
-            lambda t: t.date == self.date
-            and t.project_id.id == self.project_id.id
-            and t.task_id.id == self.task_id.id
-        )
-        new_ts = timesheets.filtered(lambda t: t.name == empty_name)
-        amount = sum(t.unit_amount for t in timesheets)
-        diff_amount = self.unit_amount - amount
-        if not diff_amount:
-            return
-        if new_ts:
-            if len(new_ts) > 1:
-                new_ts = new_ts.merge_timesheets()
-            unit_amount = new_ts.unit_amount + diff_amount
-            new_ts.write({'unit_amount': unit_amount})
-        else:
-            new_ts_values = self._line_to_timesheet(diff_amount)
-            self.env['account.analytic.line'].create(new_ts_values)
+        self._update_line()
+
+    def _update_line(self):
+        sheet = self.sheet_id
+        if self and not self.ids:
+            line = sheet.line_ids.filtered(
+                lambda l: l.date == self.date
+                and l.project_id.id == self.project_id.id
+                and l.task_id.id == self.task_id.id
+                and l.id != self.id
```

Comment:
`if not self.ids` ... and then ... `l.id != self.id` :thinking:

Reply:
hehe this case is when you have `self.id = NewId...`, so self is True, but self.ids is [], and in this filter you want to exclude this id, and find an already existing id. I have proved that while debugging, that you may have an existing sheet, but when you fill a cell, you are not editing the corresponding line but you are creating a new line. And when you have two line to same cell problems may arise, you with this method I am assuring you only have a line per cell.

Reply:
Would be more clear written this way:

```python
if isinstance(self.id, models.NewId):```

Reply:
niiiice